numba/tests/test_typedlist.py

```diff
@@ -530,6 +530,16 @@ def test_repr(self):
         expected = "ListType[int32]([1, 2, 3])"
         self.assertEqual(expected, repr(l))
 
+    def test_repr_for_ipython_shells(self):
+        try:
+            get_ipython
```

Comment:
Forgive me if I missed something here, but why is this line needed?

Reply:
Thanks for your review. From what I understood, we wanted to trim the `__str__` for large typed Lists _only_ when invoked from an `ipython` shell. `get_ipython` is only available (I think) when invoked from an `ipython` shell, otherwise a `NameError` is raised, which is an indicator that we should fall back to the original `__str__`, i.e. printing the entire typed List no matter the length. Does that make sense?

Reply:
I see, that is interesting. I wonder why this isn't being flagged by flake8?